### ISSUE TITLE
Dropout not handled properly in `scaled_dot_product_attention`?

### ISSUE THREAD

**Issue by zjost** [pytorch/torchtune#1791]
In the `MultiHeadAttention` module, the docstrings [say this](https://github.com/pytorch/torchtune/blob/57ab583c84c4a9dcacac23aeabc81f2a679670fe/torchtune/modules/attention.py#L72-L74) about `attn_dropout`:

> This argument is ignored if the self.training is False. Default value is 0.0.

But the [actual code](https://github.com/pytorch/torchtune/blob/57ab583c84c4a9dcacac23aeabc81f2a679670fe/torchtune/modules/attention.py#L297-L304) doesn't handle this, and simply passes it on:
```
output = self._attention_call(
            q,
            k,
            v,
            mask=mask,
            dropout_p=self.attn_dropout,
            is_causal=self.kv_cache is None and mask is None and self.is_causal,
        )
```
to `nn.functional.scaled_dot_product_attention`, which has this [known behavior](https://github.com/pytorch/pytorch/pull/126294).

Is this handled somewhere else that I haven't realized?

**Comment by ebsmothers** [comment]
Oh this is a good point. At least looking at the blame from when that comment was originally added, it seems that the implementation at that time did actually set `dropout_p=self.attn_dropout if self.training else 0.0` in the SDPA call (see #8). I see we updated in #45, but I think the docstring was never updated to reflect this. So we should delete this line from the docstring. (I guess a separate question is whether it's the right thing to always pass attn_dropout through like this, but in my mind it's better than sneakily modifying things under the hood)

**Comment by zjost** [comment]
I think it's actually a bug.  If I understand correctly, then there's no way to turn off dropout, which is not the desired behavior during inference.  According to https://github.com/pytorch/pytorch/pull/126294, it's up to the client to handle the training/eval logic to switch this on and off since it's a function.  

What's the concern with the old way of `dropout_p=self.attn_dropout if self.training else 0.0`?

**Comment by zjost** [comment]
Addressed in #1803 (thanks!)
